app/services/crawler.rb

```diff
@@ -0,0 +1,42 @@
+class Crawler
+  
+  def initialize(url)
+    require 'nokogiri'
+    require 'open-uri'
+    @page = Nokogiri::HTML(open(url, headers).read)
+  end
+
+  def subject
+    @page.css("title").first.text
+  end
```

Comment:
`subject`, `summary` 要寫 guard clause 以防網頁沒有 title 或 meta description。目前的寫法如果沒有遇到前述狀況會炸掉

 ```suggestion
  return if @page.css('title').empty?
  ... your code ...
```

Reply:
對耶，忽略了
但會有網頁無title嗎？

Reply:
規劃要想到 edge cases

Reply:
如果你想要你的 app 再友善一點，甚至可以在遇到沒有 title 或 summary 時候主動提醒使用者這樣的狀況